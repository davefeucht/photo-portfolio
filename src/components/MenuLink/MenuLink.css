.menu-link {
    color: var(--template-white);
}

.menu-link:visited {
    color: var(--template-white);
}
