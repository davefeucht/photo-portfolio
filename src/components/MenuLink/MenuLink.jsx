/*****************
* MenuLink component implements the a link in the menu of the application
*****************/

import React from 'react';
import PropTypes from 'prop-types';
import { Link, useRouteMatch } from 'react-router-dom';
import './MenuLink.css';

const MenuLink = ({ href, text }) => {
    let { url } = useRouteMatch();
    return (
        <Link to={`${url}${href}`}>
            <div className="menu-link">{text}</div>
        </Link>
    );
};

MenuLink.displayName = 'MenuLink';

MenuLink.propTypes = {
    href: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
};

export default MenuLink;
