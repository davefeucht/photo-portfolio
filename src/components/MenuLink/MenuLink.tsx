/** ***************
* MenuLink component implements the a link in the menu of the application
**************** */

import './MenuLink.css';

import * as React from 'react';
import { Link } from 'react-router-dom';

interface MenuLinkProps {
    href: string,
    text: string
}

const MenuLink: React.FC<MenuLinkProps> = ({ href, text }) => {
    return (
        <Link to={`${href}`}>
            <div className="menu-link">{text}</div>
        </Link>
    );
};

MenuLink.displayName = 'MenuLink';

export default MenuLink;
