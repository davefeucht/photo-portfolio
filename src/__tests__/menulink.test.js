import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import MenuLink from '../components/MenuLink/MenuLink';

const href = '/page/150';
const text = 'Contact'

test('MenuLink displays', async () => {
    let component;
    let tree;
    component = renderer.create(
        <MemoryRouter>
            <MenuLink href={href} text={text}></MenuLink>
        </MemoryRouter>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
