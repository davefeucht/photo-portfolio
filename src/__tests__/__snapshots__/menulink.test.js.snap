// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLink displays 1`] = `
<a
  href="/page/150"
  onClick={[Function]}
>
  <div
    className="menu-link"
  >
    Contact
  </div>
</a>
`;
